### PR TITLE
chore(types): [WIP] PayloadAttributes generic

### PR DESCRIPTION
**Pull request by itsdevbear**
## Summary by CodeRabbit

- **New Features**
  - Introduced `PayloadAttributes` parameter across various blockchain functions, enhancing payload handling and processing.

- **Refactor**
  - Updated function signatures and types across multiple modules to incorporate `PayloadAttributes`, improving the consistency and scalability of the codebase.

### CODE REVIEW

File: mod/beacon/blockchain/service.go [hunk 1]
```diff
@@ -109,7 +111,8 @@ func NewService[
 	],
 	logger log.Logger[any],
 	cs common.ChainSpec,
-	ee ExecutionEngine,
+	//nolint:lll // annoying formatter.
+	ee ExecutionEngine[*engineprimitives.PayloadAttributes[*engineprimitives.Withdrawal]],
```
Comment (archbear) [inline mod/beacon/blockchain/service.go:115]: is there a reason we aren't propagating the generic all the way to the service definition, this feels like one of those moments where it'll bite us back later

**Review by archbear — COMMENTED** [review]
nits that i think are important

File: mod/beacon/blockchain/service.go (hunk 1 above)
Comment (itsdevbear) [reply mod/beacon/blockchain/service.go:115]: cause i was lazy and only got halfway thru haha

File: mod/engine-primitives/pkg/engine-primitives/attributes.go [hunk 2]
```diff
@@ -92,6 +92,27 @@ func NewPayloadAttributes[
 	return p, nil
 }
 
+// New empty PayloadAttributes.
+func (p *PayloadAttributes[WithdrawalT]) New(
```
[the remark this answers is not in the document]
Comment (itsdevbear) [reply mod/engine-primitives/pkg/engine-primitives/attributes.go:96]: yeah in the interface, i wanna kill engineprimitives.Attributer tho

File: mod/beacon/blockchain/service.go (hunk 1 above)
Comment (archbear) [reply mod/beacon/blockchain/service.go:115]: LOL all good

**Pull request merged**